R/utils-ui.R

```diff
@@ -1,6 +1,118 @@
 sq <- function(x) glue::single_quote(x)
 bt <- function(x) glue::backtick(x)
 
+# message <- function(...) {
+#   abort("Internal error: use googledrive's UI functions, not {bt('message()')}")
+# }
+
+warn_for_verbose <- function(verbose = TRUE, env = parent.frame()) {
+  if (isTRUE(verbose)) {
+    return(invisible())
+  }
+
+  called_from <- sys.parent(1)
+  if (called_from == 0) {
+    # called from global env, presumably in a test or during development
+    called_as <- "some_googledrive_function()"
+  } else {
+    called_as <- deparse(sys.call(called_from))
+  }
```

Comment:
Is this the right way to do this? I don't think there is any non-deprecated rlang function that is more suited to this, but I'd love your confirmation of that @lionel- .

Staying true to my brand, I do want to be able to call this function from `globalenv()`, just because it's convenient. But "in production", it's always called from some user-facing googledrive function.

Reply:
This looks good. However I wouldn't deparse the call and then reparse it with a regexp because that is brittle and more convoluted. Instead you could:

- Check that the result of `sys.call()` is a call (just to be defensive, in some cases/versions of R it sometimes returns a literal promise object) and that the first element of the call is a symbol (it could be another call or an inlined function).
- If the checks agree, convert the first element to string and use that. Otherwise use the default string.

Reply:
Or match the lifecycle syntax a little more, and require the caller to specify the function name:

```R
warn_for_verbose(verbose, "my_function")
```

If you did want keep automatically guessing the function name, you could also lean into the call data structure and use that to generate the whole string:

```R
called_from <- sys.parent(1)
if (called_from == 0) {
  # called from global env, presumably in a test or during development
  called_as <- quote(some_googledrive_function())
} else {
  called_as <- sys.call(called_from)
}

called_as[[2]] <- quote(verbose)
caller <- as.character(called_as)
```